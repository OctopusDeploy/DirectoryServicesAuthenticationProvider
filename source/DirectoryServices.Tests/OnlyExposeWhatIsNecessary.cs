using System;
using System.Linq;
using Assent;
using NUnit.Framework;
using NUnit.Framework.Internal;
using Octopus.Server.Extensibility.Authentication.DirectoryServices;

namespace DirectoryServices.Tests
{
    [TestFixture]
    public class OnlyExposeWhatIsNecessary
    {
        [Test]
        public void ServerExtensionsShouldMinimiseWhatIsExposed()
        {
            var assembly = typeof(DirectoryServicesExtension).Assembly;

            var publicThings = assembly.GetExportedTypes()
                .Select(t => t.FullName);
            
            this.Assent(string.Join(Environment.NewLine, publicThings));
        }
    }
}