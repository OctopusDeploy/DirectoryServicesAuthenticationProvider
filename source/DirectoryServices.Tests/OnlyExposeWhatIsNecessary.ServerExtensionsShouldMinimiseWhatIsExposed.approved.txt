Octopus.Server.Extensibility.Authentication.DirectoryServices.DirectoryServicesExtension
Octopus.Server.Extensibility.Authentication.DirectoryServices.DirectoryServicesServiceCollectionContributor
Octopus.Server.Extensibility.Authentication.DirectoryServices.IntegratedAuthentication.IIntegratedAuthenticationHandler
Octopus.Server.Extensibility.Authentication.DirectoryServices.IntegratedAuthentication.IntegratedAuthenticationController
Octopus.Server.Extensibility.Authentication.DirectoryServices.IntegratedAuthentication.ApiConstants
Octopus.Server.Extensibility.Authentication.DirectoryServices.IntegratedAuthentication.DomainCookieOptions
Octopus.Server.Extensibility.Authentication.DirectoryServices.Configuration.DirectoryServicesConfiguration
Octopus.Server.Extensibility.Authentication.DirectoryServices.Configuration.IDirectoryServicesConfigurationStore